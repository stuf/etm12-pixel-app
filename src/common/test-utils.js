import * as R from 'ramda';
import * as V from 'partial.lenses.validation';
