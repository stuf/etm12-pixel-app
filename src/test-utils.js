/**
 * @module tests
 *
 * Contains some common test utility functions for use in
 * unit tests.
 *
 * Taken from {@link https://github.com/calmm-js/karet.util/blob/master/test/tests.js}
 */
import { Observable, Property, Stream, constant as C } from 'kefir';
import * as U from 'karet.util';
import * as R from 'kefir.ramda';

export function show(x) {
  switch (typeof x) {
    case 'string':
    case 'object':
      return JSON.stringify(x);
    default:
      return `${x}`;
  }
}

export const toExpr = f =>
  f
    .toString()
    .replace(/\s+/g, ' ')
    .replace(/^\s*function\s*\(\s*\)\s*{\s*(return\s*)?/, '')
    .replace(/\s*;?\s*}\s*$/, '')
    .replace(/function\s*(\([a-zA-Z]*\))\s*/g, '$1 => ')
    .replace(/{\s*return\s*([^{;]+)\s*;\s*}/g, '$1')
    .replace(/\(([a-zA-Z0-9_]+)\) =>/g, '$1 =>')
    .replace(/\(0, _kefir[^.]*[.]constant\)/g, 'C')
    .replace(/_kefir[^.][.]/g, '')
    .replace('() => ', '');

export const testEq = (expect, thunk) =>
  it(`${toExpr(thunk)} => ${show(expect)}`, done => {
    const actual = thunk();
    const check = actual => {
      const eq = R.equals(actual, expect);
      if (eq instanceof Observable || !eq) {
        done(Error(`Expected: ${show(expect)}, actual: ${show(actual)}`));
      } else {
        done();
      }
    };
    if (actual instanceof Observable) {
      U.thru(actual, U.takeFirst(1), U.on({ value: check, error: check }));
    } else {
      check(actual);
    }
  });

export const testThrows = thunk =>
  it(`${toExpr(thunk)} => throws`, () => {
    try {
      thunk();
    } catch (_) {
      return;
    }
    throw Error('Did not throw as expected.');
  });

export const testRender = (expect, vdomThunk) =>
  it(`${expect}`, () => {
    const actual = ReactDOM.renderToStaticMarkup(vdomThunk());

    if (actual !== expect)
      throw new Error(`Expected: ${show(expect)}, actual: ${show(actual)}`);
  });
