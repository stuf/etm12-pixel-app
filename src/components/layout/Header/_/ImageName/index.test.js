import * as React from 'karet';
import { shallow, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as K from 'kefir';

import { testEq } from 'test-utils';

import Index from './index';

it('snapshot test', () => {
  const w = shallow(<Index />);
  expect(toJson(w)).toMatchSnapshot('with empty data');
});

it('displays correct element based on isEditing prop', () => {
  const w = shallow(<Index {...{ isEditing: false, name: 'Untitled 123' }} />);
  expect(w.find('.nameLabel').text()).toBe('Untitled 123');

  const w2 = render(<Index {...{ isEditing: true, name: 'Untitled 123' }} />);
  expect(w2.find('.nameLabel')).toHaveLength(0);
  expect(w2.find('input')).not.toHaveLength(0);
});
