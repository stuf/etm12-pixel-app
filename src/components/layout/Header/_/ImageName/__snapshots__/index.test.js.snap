// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="root flex-horizontal -vertical-center"
>
  <span
    className="nameLabel"
    onClick={[Function]}
  />
</div>
`;

exports[`snapshot test: with empty data 1`] = `
<div
  className="root flex-horizontal -vertical-center"
>
  <span
    className="nameLabel"
    onClick={[Function]}
  />
</div>
`;
